spring-xd-shell/src/test/java/org/springframework/xd/shell/command/FileSourceAndFileSinkTests.java

```diff
@@ -113,4 +120,38 @@ private void testFileSinkSource(String sinkSuffix, String sinkParam) throws Exce
 			out.delete();
 		}
 	}
+
+	private void testFileSinkSource(String nameExpressionParam, String dirPrefix, String dirSuffix) throws Exception {
+		// Are we on *nix at least?
+		if (!new File("/tmp/").exists()) {
+			return;
+		}
+
+		// Both use stream name
+		String streamName = generateStreamName();
+
+		File inDir = new File(DEFAULT_IN, streamName);
+		inDir.mkdirs();
+		File in = new File(inDir, "one.txt");
+		File outDir = new File("/tmp/" + dirPrefix + "hello" + dirSuffix);
+		File out = new File(outDir, "hello");
+
+		try {
+			FileCopyUtils.copy("hello", new FileWriter(in));
+			String directoryExpression = " --directoryExpression='''/tmp/'' + ''" + dirPrefix + "'' + payload.trim() + ''"
+					+ dirSuffix + "'''";
+			stream().create(streamName, "in: file --outputType=text/plain | out: file " + nameExpressionParam +
+					directoryExpression);
+			Thread.sleep(1000);
```

Comment:
Also, arbitrary sleeps like this are unreliable on the CI servers and can extend the overall test times; it's better to intelligently wait for some condition (e.g. loop 100 times for 100ms until the condition is true).

It provides a fast exit when the condition is generally true faster than the arbitrary sleep, while allowing for a longer time when running on a heavily loaded server.